File: src/dataset/go/sublist_Exercism_go_rootulp/gemini/GeminiFlash_sublist_v1.go
package sublist

type Relation string

func Sublist(a []int, b []int) Relation {
	lenA := len(a)
	lenB := len(b)

	if lenA == 0 && lenB == 0 {
		return "equal"
	}

	if lenA == 0 {
		return "sublist"
	}

	if lenB == 0 {
		return "superlist"
	}

	if lenA == lenB {
		if equal(a, b) {
			return "equal"
		}
	} else if lenA < lenB {
		if isSublist(a, b) {
			return "sublist"
		}
	} else {
		if isSublist(b, a) {
			return "superlist"
		}
	}

	return "unequal"
}

func equal(a, b []int) bool {
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

func isSublist(a []int, b []int) bool {
	lenA := len(a)
	lenB := len(b)

	if lenA > lenB {
		return false
	}

	for i := 0; i <= lenB-lenA; i++ {
		if equal(a, b[i:i+lenA]) {
			return true
		}
	}

	return false
}

File: src/dataset/c/sublist_Exercism_m3g4d1v3r/gemini/GeminiFlash_sublist_v1.c
#include "sublist.h"
#include <stdbool.h>

comparison_result_t check_lists(int *list_a, int *list_b, size_t len_a,
                                size_t len_b) {
    if (len_a == 0 && len_b == 0) return EQUAL;
    if (len_a == 0) return SUBLIST;
    if (len_b == 0) return SUPERLIST;

    if (len_a == len_b) {
        for (size_t idx = 0; idx < len_a; idx++) {
            if (list_a[idx] != list_b[idx]) return UNEQUAL;
        }
        return EQUAL;
    }

    comparison_result_t target_result;
    int *sml_list, *big_list;
    size_t sml_len, big_len;

    if (len_a < len_b) {
        target_result = SUBLIST;
        sml_list = list_a;
        sml_len = len_a;
        big_list = list_b;
        big_len = len_b;
    } else {
        target_result = SUPERLIST;
        sml_list = list_b;
        sml_len = len_b;
        big_list = list_a;
        big_len = len_a;
    }

    if (sml_len > big_len) return UNEQUAL;

    for (size_t i = 0; i <= big_len - sml_len; ++i) {
        bool found = true;
        for (size_t j = 0; j < sml_len; ++j) {
            if (big_list[i + j] != sml_list[j]) {
                found = false;
                break;
            }
        }
        if (found) {
            return target_result;
        }
    }

    return UNEQUAL;
}